editor/plugins/script_text_editor.cpp

```diff
@@ -1612,10 +1614,20 @@ void ScriptTextEditor::_color_changed(const Color &p_color) {
 		new_args = String("(" + rtos(p_color.r) + ", " + rtos(p_color.g) + ", " + rtos(p_color.b) + ", " + rtos(p_color.a) + ")");
 	}
 
-	String line = code_editor->get_text_edit()->get_line(color_line);
-	String new_line = line.replace(color_args, new_args);
+	String line_with_replaced_args = _create_line_with_replaced_color_args(new_args);
+
 	color_args = new_args;
-	code_editor->get_text_edit()->set_line(color_line, new_line);
+	code_editor->get_text_edit()->set_line(color_position.x, line_with_replaced_args);
+	code_editor->get_text_edit()->update();
+}
+
+String ScriptTextEditor::_create_line_with_replaced_color_args(String new_args) {
```

Comment:
I don't think this needs to be a separate method if it's only used in one place. If the code needs more documentation, you can add a comment to say that it "creates line with replaced color args" directly.

Reply:
Otherwise the rest seems good, so it should be good to merge once that's fixed.

Reply:
method removed